compiler/src/main/java/com/squareup/anvil/compiler/AnvilComponentRegistrar.kt

```diff
@@ -91,42 +91,19 @@ class AnvilComponentRegistrar : ComponentRegistrar {
     project: MockProject,
     extension: AnalysisHandlerExtension
   ) {
-    @Suppress("DEPRECATION")
-    val analysisHandlerExtensionPoint = Extensions.getArea(project)
+    val analysisHandlerExtensionPoint = project.extensionArea
         .getExtensionPoint(AnalysisHandlerExtension.extensionPointName)
+        .let {
+          it as? ExtensionPointImpl ?: throw AnvilCompilationException(
+              "Expected AnalysisHandlerExtension to be an instance of ExtensionPointImpl. The " +
+                  "class is ${it::class.java}"
+          )
+        }
 
-    val registeredExtensions = AnalysisHandlerExtension.getInstances(project)
-    registeredExtensions.forEach {
-      // This doesn't work reliably, but that's the best we can do with public APIs. There's a bug
-      // for inner classes where they convert the given class to a String "a.b.C.Inner" and then
-      // try to remove "a.b.C$Inner". Good times! Workaround is below.
-      analysisHandlerExtensionPoint.unregisterExtension(it::class.java)
-    }
-
-    if (analysisHandlerExtensionPoint.hasAnyExtensions() &&
-        analysisHandlerExtensionPoint is ExtensionPointImpl<AnalysisHandlerExtension>
-    ) {
-      AnalysisHandlerExtension.getInstances(project)
-          .forEach {
-            analysisHandlerExtensionPoint.unregisterExtensionFixed(it::class.java)
-          }
-    }
-
-    check(!analysisHandlerExtensionPoint.hasAnyExtensions()) {
-      "There are still registered extensions."
-    }
-
-    AnalysisHandlerExtension.registerExtension(project, extension)
-    registeredExtensions.forEach { AnalysisHandlerExtension.registerExtension(project, it) }
-  }
-
-  private fun <T : AnalysisHandlerExtension> ExtensionPointImpl<T>.unregisterExtensionFixed(
-    extensionClass: Class<out T>
-  ) {
-    // The bug is that they use "extensionClass.canonicalName".
-    val classNameToUnregister = extensionClass.name
-    unregisterExtensions({ className, _ ->
-      classNameToUnregister != className
-    }, true)
+    ExtensionPointImpl::class.java.declaredMethods
+        .first { it.name == "doRegisterExtension" }
+        .use {
+          it.invoke(analysisHandlerExtensionPoint, extension, LoadingOrder.FIRST, project)
+        }
```

Comment:
I called this out in the commit message. I'm not concerned about this workaround at all, because there is supposed to be a public API for that, but I can't see it yet. This API is actually part of the IntelliJ platform. The Kotlin compiler repackages and bundles the code. But this one API is missing either because it's an older version or the method is stripped. JetBrains actually suggested in the ticket to use the non-existent API https://youtrack.jetbrains.com/issue/KT-42103

Reply:
That comment might be helpful to actually have as a comment in the source code.

Reply:
Yep, good idea.